### PR TITLE
Merge system certificate pool with custom certificates

### LINKED ISSUE #12756: System CA pool excluded when registry CA is used from /etc/docker
Currently if a user specifies a CA for a private registry in `/etc/docker/certs.d/REGISTRY/ca.crt`, the TLS configuration is configured to use a new CA pool with only that CA. If the registry does a redirect to a URL which is under a new domain with a certificate valid using the system CA pool but not the specified private registry CA, the operation will fail as an unknown certificate. The expected behavior would be to add the registry CA to the list of trusted roots along with the system roots.

See also distribution issue https://github.com/docker/distribution/issues/426

@ping moxiegirl @docker/distribution-maintainers

### PR DESCRIPTION
**Pull request by dmcgowan**
Updates `tlsconfig` from go connections to merge custom certificates with the system. Interactions with the registry will make use of this new call to get the system certificates.

Additionally removed deprecated `pkg/tlsconfig/config.go`. This code has already been migrated to go-connections, was out dated and no longer imported.

Closes #12756

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Comment by thaJeztah** [comment]
Do we need documentation changes for this? IIRC we had a section on where to put the certificates

**Comment by dmcgowan** [comment]
@thaJeztah yes, I think we need to update the note about it here https://docker.github.io/engine/security/certificates/#/understanding-the-configuration

File: vendor/src/github.com/docker/go-connections/tlsconfig/certpool_other.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// +build !go1.7
```
Comment (tiborvass) [inline vendor/src/github.com/docker/go-connections/tlsconfig/certpool_other.go]: @dmcgowan just curious, how will this be resolved in go1.8 ?

Comment (dmcgowan) [inline vendor/src/github.com/docker/go-connections/tlsconfig/certpool_other.go]: From https://golang.org/pkg/go/build/
> - "go1.7", from Go version 1.7 onward

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**